### PR TITLE
[User Model ]Remove early return of trackReceivedEvent to allow for os_notification_influence_open event to be sent to Firebase

### PR DESCRIPTION
**Pull request by emawby**
# Description
## One Line Summary
Bringing #1241 To user model branch

## Details
Cherry pick of #1241 
Also adding crash protection against a nil notificaiton id and campaign

## Manual testing

# Affected code checklist
   - [x] Notifications
      - [ ] Display
      - [ ] Open
      - [ ] Push Processing
      - [ ] Confirm Deliveries
   - [ ] Outcomes
   - [ ] Sessions
   - [ ] In-App Messaging
   - [ ] REST API requests
   - [ ] Public API changes

# Checklist
## Overview
   - [x] I have filled out all **REQUIRED** sections above
   - [x] PR does one thing
     - If it is hard to explain how any codes changes are related to each other then it most likely needs to be more than one PR
   - [x] Any Public API changes are explained in the PR details and conform to existing APIs

## Testing
   - [x] I have included test coverage for these changes, or explained why they are not needed
   - [x] All automated tests pass, or I explained why that is not possible
   - [x] I have personally tested this on my device, or explained why that is not possible

## Final pass
   - [x] Code is as readable as possible.
      - Simplify with less code, followed by splitting up code into well named functions and variables, followed by adding comments to the code.
   - [x] I have reviewed this PR myself, ensuring it meets each checklist item
      - WIP (Work In Progress) is ok, but explain what is still in progress and what you would like feedback on. Start the PR title with "WIP" to indicate this.

- - -
This change is [Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-iOS-SDK/1296)

### CODE REVIEW

File: iOS_SDK/OneSignalSDK/OneSignalCore/Source/OneSignalTrackFirebaseAnalytics.m [hunk 1]
```diff
@@ -145,6 +134,10 @@ + (void)trackInfluenceOpenEvent {
     NSString *notificationId = [sharedUserDefaults getSavedStringForKey:ONESIGNAL_FB_LAST_NOTIFICATION_ID_RECEIVED defaultValue:nil];
     NSString *campaign = [sharedUserDefaults getSavedStringForKey:ONESIGNAL_FB_LAST_GAF_CAMPAIGN_RECEIVED defaultValue:nil];
     
+    if (!notificationId || !campaign) {
```
Comment (jkasten2) [inline iOS_SDK/OneSignalSDK/OneSignalCore/Source/OneSignalTrackFirebaseAnalytics.m:137]: Campaigns are mapped to the OneSignal template_id, they are not set a lot of the times, however it is not require by Firebase so we should still allow the event to go through so it isn't skipped completely.

Comment (emawby) [reply iOS_SDK/OneSignalSDK/OneSignalCore/Source/OneSignalTrackFirebaseAnalytics.m:137]: If campaign is nil the app will crash so we cannot allow nil

Comment (emawby) [reply iOS_SDK/OneSignalSDK/OneSignalCore/Source/OneSignalTrackFirebaseAnalytics.m:137]: https://github.com/OneSignal/OneSignal-iOS-SDK/issues/1253

Comment (jkasten2) [reply iOS_SDK/OneSignalSDK/OneSignalCore/Source/OneSignalTrackFirebaseAnalytics.m:137]: Can we conditionally create the `campaign` key if it is not `nil`?

Comment (emawby) [reply iOS_SDK/OneSignalSDK/OneSignalCore/Source/OneSignalTrackFirebaseAnalytics.m:137]: We need to make this fix on main as well

**Review by jkasten2 — APPROVED** [review]

File: iOS_SDK/OneSignalSDK/OneSignalCore/Source/OneSignalTrackFirebaseAnalytics.m (hunk 1 above)
Comment (jkasten2) [reply iOS_SDK/OneSignalSDK/OneSignalCore/Source/OneSignalTrackFirebaseAnalytics.m]: agreed, we probably have this same bug

**Pull request merged**